### DISCUSSION TITLE
Docker buildx build image already created don't pull correctly

### DISCUSSION THREAD

**Discussion by mamu-reply** [docker/buildx#2418] [Q&A]
Hi,
I have this situation, I built a dockerfile, then I created an instance for buildx and subsequently I built another dockerfile with buildx which has as FROM the image I had built previously.
E.g.
- Dockefile1:
  ```
  FROM nginx
  RUN echo "test1"
  ```

- ```docker build -t test test1/Dockerfile1```
- ```docker buildx create test --use```

- Dockerfile2
  ```
  FROM test
  RUN echo "test2"
  ```

- ```docker buildx build -t test2 test2/Dockerfile2```

and i have this error

`ERROR: failed to solve: test: failed to resolve source metadata for docker.io/library/test:latest: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed`

But if i do the same thing without the buildx create command it work fine, but i need this commando to set a driver.
Anyone can help me? Thanks

**Comment by richardthombs** [comment]
I encountered a similar error and it was because we had created a layer called "tests" but were mistakenly referring to it as "test" when copying from it. If you try to pull from a non-existent layer called "foobar" for example:

![image](https://github.com/user-attachments/assets/5f3ca23c-9b17-4688-8c21-63b4fcfd2e64)

**Comment by ap-wtioit** [comment]
This also affects docker compose (the plugin) for local only images (e.g. in tests): https://github.com/docker/compose/issues/9939

Edit:
to solve this i needed to re-create the buildx instance:
```bash
docker buildx rm --all-inactive
docker buildx create --driver docker --use default # for me this gave an error but still worked
docker buildx inspect # should show Driver: docker
```
see https://github.com/docker/compose/issues/9939#issuecomment-1542214481 and https://github.com/docker/buildx/issues/159#issuecomment-539238262

**Comment by bernard-code-lab** [comment]
Yeah, figured out what's happening here. When you run docker buildx create test --use, it spins up a docker-container driver by default, which basically isolates BuildKit in its own container. Local images you built with docker build don't exist inside that isolated container.

So when BuildKit tries to resolve FROM test, it can't find anything locally and attempts to pull from docker.io/library/test instead — that's where your 407 is coming from.

Few ways to fix it:

```
docker buildx build --load -t test2 .
```

This forces docker driver for that build, so you get access to local images.

If you want the builder itself to use docker driver:

```
docker buildx rm test
docker buildx create --driver docker --use
```

Tradeoff is you lose the isolation and multi-platform builds get messy.

If you need the docker-container driver (for multi-arch):
Just push to a local registry or push directly:

```
docker buildx build --push -t localhost:5000/test2 .
```

BuildKit can access it that way.

What are you trying to build — just local stuff or actually going multi-platform?
